RoddyCore/src/de/dkfz/roddy/tools/ComplexLine.groovy

```diff
@@ -0,0 +1,196 @@
+    static Map<Character, Character> possibleContainersByClosingCharacter = [:]
+
+    static {
+        possibleContainersByOpeningCharacter['\'' as char] = '\'' as Character
+        possibleContainersByOpeningCharacter['\"' as char] = '\"' as Character
+        possibleContainersByOpeningCharacter['{' as char] = '}' as Character
+        possibleContainersByOpeningCharacter['[' as char] = ']' as Character
+        possibleContainersByOpeningCharacter['(' as char] = ')' as Character
+
+        possibleContainersByOpeningCharacter.each { Character key, Character value -> possibleContainersByClosingCharacter[value] = key }
+    }
+
+    String line;
+    LineNode parsedLine;
+
+    /**
+     * Parsed line represents a transformed / parsed line. Currently, single and double ticks will be used
+     * to identify children.
+     *
+     * E.g. --abc --def="ab,def;efk=test;anotherValue='equals'" --jki --a="something=def"
+     *
+     * will be represented like:
+     *
+     *     --abc --def="###CHILD_0###" --jki --a="###CHILD_1###"
+     *                       |                         |
+     *                       ab,def;efk=test;anotherValue="###CHILD_0###"
+     *                                                 |        |
+     *                                                 something=def
+     *                                                          |
+     *                                                          equals
+     */
+    static abstract class LineNode {
+
+        abstract String reassemble()
+
+        abstract String[] splitBy(String s)
+
+        @Override
+        public String toString() {
+            return reassemble()
+        }
+    }
+
+    static class TextNode extends LineNode {
+        StringBuilder text = new StringBuilder()
+
+        TextNode() {
+
+        }
+
+        TextNode(String start) {
+            this.text << start
+        }
+
+        @Override
+        String reassemble() {
+            return text.toString()
+        }
+
+        @Override
+        String[] splitBy(String s) {
+            return text.toString().split(s)
+        }
+    }
+
+    static class ComplexNode extends LineNode {
+        List<LineNode> blocks = []
+
+        ComplexNode() {
+            blocks << new TextNode()
+        }
+
+
+        @Override
+        String reassemble() {
+            StringBuilder _content = new StringBuilder();
+            for (int i = 0; i < blocks.size(); i++) {
+                _content << blocks[i].reassemble()
+            }
+            return _content.toString();
+        }
+
+        public String[] splitBy(String s) {
+            // Keep a list of splitted, prepared blocks. The list will be used to join results.
+            List<StringBuilder> splittedBlocks = []
+            for (int i = 0; i < blocks.size(); i++) {
+                List<String> splitted = null
+                if (blocks[i] instanceof TextNode) {
```

Comment:
Hm, would it be more appropriate to use a intuitively named, polymorphic (dynamic) method on TextNode and ComplexNode, that does the right thing, instead of the "instanceof"?